internal/controller/utils.go

```diff
@@ -74,3 +78,22 @@ func paramsToEnv(params map[string]intstr.IntOrString) []corev1.EnvVar {
 	}
 	return envs
 }
+
+func CreateKubernetesClient() (*kubernetes.Clientset, error) {
+	// Try in-cluster config
+	config, err := rest.InClusterConfig()
```

Comment:
No need to do this, grab this from `mgr.GetConfig()` inside of `main()`.

Reply:
fixed, big improvement to do it this way. Thanks for catching these  kind of improvements.